src/docfx/lib/github/GitHubAccessor.cs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Octokit;
+
+namespace Microsoft.Docs.Build
+{
+    internal static class GitHubAccessor
+    {
+        private static readonly GitHubClient _client = new GitHubClient(new ProductHeaderValue("DocFXv3"));
+
+        // 0 for false, 1 for true.
+        private static int _isRateLimitExceeded = 0;
```

Comment:
>int [](start = 23, length = 3)

how about `bool`?

Reply:
`Interlocked.Exchange` doesn't support bool. It's sample on docs also use int for bool.

---
In reply to: [207451563](https://github.com/dotnet/docfx/pull/3144#discussion_r207451563) [](ancestors = 207451563)

Reply:
Use `Interlocked` to ensure I only log one warning for exceeding rate limit before. Remove this now as the warning is embedded in authorNotFound warning.

---
In reply to: [207452556](https://github.com/dotnet/docfx/pull/3144#discussion_r207452556) [](ancestors = 207452556,207451563)